diff -r 5879929b0103 web/org.openbravo.mobile.core/source/retail/component/ob-retail-searchproductcharacteristic.js
--- a/web/org.openbravo.mobile.core/source/retail/component/ob-retail-searchproductcharacteristic.js	Tue Mar 10 09:58:46 2015 +0530
+++ b/web/org.openbravo.mobile.core/source/retail/component/ob-retail-searchproductcharacteristic.js	Tue Mar 10 16:28:30 2015 +0530
@@ -49,7 +49,8 @@
         components: [{
           kind: 'OB.UI.SearchInputAutoFilter',
           name: 'productname',
-          style: 'width: 100%;'
+          style: 'width: 100%;',
+          minLengthToSearch: 4
         }]
       }, {
         style: 'display: table-cell;',
@@ -313,7 +314,8 @@
   }, {
     style: 'display: table; width:100%',
     components: [{
-      style: 'display: table-cell; width:30%',
+      name: 'characteristicsFilterContainer',
+      style: 'width:30%',
       classes: 'row-fluid',
       components: [{
         components: [{
@@ -355,6 +357,11 @@
     this.$.products.setCollection(this.products);
     this.$.productsCh.setCollection(this.productsCh);
     //    this.$.products.getHeader().setHeaderCollection(this.categories);
+    //preference
+    this.$.characteristicsFilterContainer.addStyles('display: table-cell;');
+    if (OB.MobileApp.model.hasPermission('OBPOS_HideProductCharacteristics', true)) {
+      this.$.characteristicsFilterContainer.addStyles('display:none;');
+    }
 
     function errorCallback(tx, error) {
       OB.UTIL.showError("OBDAL error: " + error);
